src/py-opentimelineio/opentimelineio/adapters/file_bundle_utils.py

```diff
@@ -103,68 +102,85 @@ def _prepped_otio_for_bundle_and_manifest(
-                    " target_url attributes that begin with 'file:'.  Got a "
-                    "target_url of:  '{}'".format(adapter_name, target_url)
-                )
-            if media_policy is MediaReferencePolicy.MissingIfNotFile:
-                cl.media_reference = reference_cloned_and_missing(
-                    cl.media_reference,
-                    "target_url is not a file scheme url (start with url:)"
-                )
-                continue
-
-        # get an absolute path to the target file
-        target_file = os.path.abspath(url_utils.filepath_from_url(target_url))
-
-        # if the file hasn't already been checked
-        if (
-            target_file not in path_to_reference_map
-            and target_file not in invalid_files
-            and (
-                not os.path.exists(target_file)
-                or not os.path.isfile(target_file)
-            )
-        ):
-            invalid_files.add(target_file)
-
-        if target_file in invalid_files:
-            if media_policy is MediaReferencePolicy.ErrorIfNotFile:
-                raise NotAFileOnDisk(target_file)
-            if media_policy is MediaReferencePolicy.MissingIfNotFile:
-                cl.media_reference = reference_cloned_and_missing(
-                    cl.media_reference,
-                    "target_url target is not a file or does not exist"
-                )
-
-                # do not need to relink it in the future or add this target to
-                # the manifest, because the path is either not a file or does
-                # not exist.
-                continue
-
-        # add the media reference to the list of references that point at this
-        # file, they will need to be relinked
-        path_to_reference_map.setdefault(target_file, []).append(
-            cl.media_reference
-        )
-
-    _guarantee_unique_basenames(path_to_reference_map.keys(), adapter_name)
-
-    return result_otio, path_to_reference_map
+        for mr in cl.media_references().values():
+            if isinstance(mr, schema.ImageSequenceReference):
+                url = cl.media_reference.target_url_base
+            else:
+                try:
+                    url = cl.media_reference.target_url
+                except AttributeError:
+                    # not an ImageSequenceReference or object with target_url,
+                    # ignoring it.
+                    continue
+
+            parsed_url = urlparse.urlparse(url)
+
+            # ensure that the urlscheme is either file or ""
+            # file means "absolute path"
+            # none is interpreted as a relative path, relative to cwd
+            if parsed_url.scheme not in ("file", ""):
+                if media_policy is MediaReferencePolicy.ErrorIfNotFile:
+                    raise NotAFileOnDisk(
+                        "The {} adapter only works with file URLs."
+                        " Got a URL of:  '{}'".format(adapter_name, url)
```

Comment:
Minor note, update to `f-string`

```suggestion
                        f"The {adapter_name} adapter only works with file URLs."
                        f" Got a URL of:  '{url}'"
```